ligolang/src/passes/13deku-zincing/compiler.ml

```diff
@@ -376,6 +390,15 @@ and compile_pattern_matching :
   let other_compile = other_compile ~raise in
   let compile_type = compile_type ~raise in
   let compiled_type = compile_type matchee.type_expression in
+  let rows =
+    matchee.type_expression.type_content |> function
+    | T_sum r | T_record r -> LMap.keys r.content
+    | T_constant const ->
+        if Ligo_string.extract const.injection = "option" then
+          [ Label "Some"; Label "None" ]
+        else []
```

Comment:
Do you know why this is necessary when we already have 

```
  | C_NONE -> Adt (MakeRecord 0) :: Adt (MakeVariant 1) :: k
  | C_SOME -> Adt (MakeVariant 0) :: k
```

on line 226?

Reply:
Good, catch I can simplify this a bit as I already know none and some order